Mathlib/RingTheory/TensorProduct/Basic.lean

```diff
@@ -948,6 +948,51 @@ theorem congr_trans [Algebra S C] [IsScalarTower R S C]
 
 theorem congr_symm (f : A ≃ₐ[S] B) (g : C ≃ₐ[R] D) : congr f.symm g.symm = (congr f g).symm := rfl
 
+variable (R A B C) in
+/-- Tensor product of algebras analogue of `mul_left_comm`. -/
+def leftComm : A ⊗[R] B ⊗[R] C ≃ₐ[R] B ⊗[R] A ⊗[R] C :=
+  let e₁ := (Algebra.TensorProduct.assoc R A B C).symm
+  let e₂ := congr (Algebra.TensorProduct.comm R A B) (1 : C ≃ₐ[R] C)
+  let e₃ := Algebra.TensorProduct.assoc R B A C
+  e₁.trans (e₂.trans e₃)
+
+@[simp]
+theorem leftComm_tmul (m : A) (n : B) (p : C) :
+    leftComm R A B C (m ⊗ₜ (n ⊗ₜ p)) = n ⊗ₜ (m ⊗ₜ p) :=
+  rfl
+
+@[simp]
+theorem leftComm_symm_tmul (m : A) (n : B) (p : C) :
+    (leftComm R A B C).symm (n ⊗ₜ (m ⊗ₜ p)) = m ⊗ₜ (n ⊗ₜ p) :=
+  rfl
+
+@[simp]
+theorem leftComm_toLinearEquiv :
+    (leftComm R A B C : _ ≃ₗ[R] _) = _root_.TensorProduct.leftComm R A B C := rfl
+
+variable (R A B C D) in
+/-- Tensor product of algebras analogue of `mul_mul_mul_comm`. -/
+def tensorTensorTensorComm : (A ⊗[R] B) ⊗[R] C ⊗[R] D ≃ₐ[R] (A ⊗[R] C) ⊗[R] B ⊗[R] D :=
```

Comment:
Do you want this for bialgebras too? (optional)

Reply:
I'm going to leave that out for now.